### PR TITLE
COMPASS-4125 - Clean up d3-tip nodes after component unmount

### PR DESCRIPTION
**Pull request by addaleax**
(This PR targets `master`, but I obviously only tested this for 2.x and will only release 2.x after this is merged.)

### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added

## Motivation and Context

- [x] Bugfix
- [ ] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [x] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/modules/date.js [hunk 1]
```diff
@@ -66,7 +67,7 @@ const minicharts_d3fns_date = (appRegistry) => {
 
   // set up tooltips
   const tip = d3.tip()
-    .attr('class', 'd3-tip')
+    .attr('class', 'd3-tip d3-tip-date')
```
Comment (addaleax) [inline src/modules/date.js:70]: The extra classes were just helpful for debugging, happy to remove them if that’s better

**Comment by mcasimir** [comment]
Amazing! Can we also put master back to the version that is used in compass and have 3.0 as separate branch?

**Comment by addaleax** [comment]
> Amazing! Can we also put master back to the version that is used in compass and have 3.0 as separate branch?

We could, but I wouldn’t mix it in with this bugfix, and it’d be a bit unfortunate because it probably really doesn’t take much to make the current master branch work

**Review by Anemy — APPROVED** [review]
lgtm!

**Pull request merged**